tower-service/src/lib.rs

```diff
@@ -201,14 +202,15 @@ pub trait Service {
     ///
     /// Calling `call` without calling `poll_ready` is permitted. The
     /// implementation must be resilient to this fact.
-    fn call(&mut self, req: Self::Request) -> Self::Future;
+    fn call(&mut self, req: Request) -> Self::Future;
 }
 
 /// Future yielding a `Service` once the service is ready to process a request
 ///
 /// `Ready` values are produced by `Service::ready`.
-pub struct Ready<T> {
+pub struct Ready<T, R> {
     inner: Option<T>,
+    _req: PhantomData<fn() -> R>,
```

Comment:
A `PhantomData` was necessary here because the impl of `Future` for `Ready` needs a `T: Service` bound, and so we need the request type for that bound.

I think this could also be resolved by adding a `T: Service<R>` bound on the definition of `Ready`?